charts/tidb-operator/templates/admission/pre-delete-job.yaml

```diff
@@ -1,106 +0,0 @@
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/component: admission-webhook
-    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+"  "_" }}
-  annotations:
-    # This is what defines this resource as a hook. Without this line, the
-    # job is considered part of the release.
-    "helm.sh/hook": pre-delete
-    "helm.sh/hook-weight": "-1"
-    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded,hook-failed
-rules:
-  - apiGroups: ["admissionregistration.k8s.io"]
-    resources: ["validatingwebhookconfigurations"]
-    verbs: ["delete"]
----
-
-kind: ClusterRoleBinding
-apiVersion: rbac.authorization.k8s.io/v1
-metadata:
-  name: {{ .Release.Name }}
-  labels:
-    app.kubernetes.io/name: {{ template "chart.name" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/component: admission-webhook
-    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+"  "_" }}
-  annotations:
-    # This is what defines this resource as a hook. Without this line, the
-    # job is considered part of the release.
-    "helm.sh/hook": pre-delete
-    "helm.sh/hook-weight": "-1"
-    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded,hook-failed
-subjects:
-  - kind: ServiceAccount
-    name: {{ .Release.Name }}
-    namespace: {{ .Release.Namespace }}
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: ClusterRole
-  name: {{ .Release.Name }}
-
----
-apiVersion: batch/v1
-kind: Job
-metadata:
-  name: webhook-del
-  namespace: {{ .Release.Namespace }}
-  labels:
-    app.kubernetes.io/name: {{ template "chart.name" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/component: admission-webhook
-    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+"  "_" }}
-  annotations:
-    # This is what defines this resource as a hook. Without this line, the
-    # job is considered part of the release.
-    "helm.sh/hook": pre-delete
-    "helm.sh/hook-weight": "-1"
-    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded,hook-failed
-spec:
-  template:
-    metadata:
-      name: webhook-del
-    spec:
-    {{- if .Values.imagePullSecrets }}
-      imagePullSecrets:
-{{ toYaml .Values.imagePullSecrets | indent 6 }}
-    {{- end }}
-      restartPolicy: Never
-      serviceAccountName: {{ .Release.Name }}
-      containers:
-        - name: "{{.Release.Name}}-job-del"
-          image: {{ .Values.admissionWebhook.jobImage }}
-          imagePullPolicy: IfNotPresent
-          command:
-            - "sh"
-            - "-c"
-            - |
-              set -e
-              kubectl delete validatingWebhookConfigurations.admissionregistration.k8s.io validation-tidb-pod-webhook-cfg || true
```

Comment:
No, tested by Helm3, and the webhook config can be normally deleted rather than pending even if we delete the job here, which is not the same as the PR https://github.com/pingcap/tidb-operator/pull/1389#issue-540852364 said. Then should I keep this?  

### step 1: Helm install tidb-operator with pod webhook enabled.
- config differs from `charts/tidb-operator/values.yaml`:
```
operatorImage: pingcap/tidb-operator:v1.2.3
admissionWebhook:
  create: true
```
- result:
```
# helm install tidb-operator charts/tidb-operator/
# kubectl get validatingWebhookConfigurations.admissionregistration.k8s.io
NAME                              WEBHOOKS   AGE
validation-tidb-pod-webhook-cfg   1          63s
# kubectl get pods --namespace default -l app.kubernetes.io/instance=tidb-operator
NAME                                       READY   STATUS    RESTARTS   AGE
tidb-admission-webhook-6cf69dcfbd-f5wpm    1/1     Running   0          84s
tidb-controller-manager-7f5fcbc9b4-5szmb   1/1     Running   0          84s
tidb-scheduler-794f885b9f-8jmgx            2/2     Running   0          84s
# helm get hooks tidb-operator
---
# Source: tidb-operator/templates/admission/pre-delete-job.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: tidb-operator
  namespace: default
  labels:
    app.kubernetes.io/name: tidb-operator
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/instance: tidb-operator
    app.kubernetes.io/component: admission-webhook
    helm.sh/chart: tidb-operator-v1-canary
...
```
### step 2: Delete file of `charts/tidb-operator/templates/admission/pre-delete-job.yaml`
### step 3: Upgrade tidb-operator to v1.2.4 with same values.yaml to step 1.the result is:
```
# helm upgrade tidb-operator charts/tidb-operator/
# kubectl get validatingWebhookConfigurations.admissionregistration.k8s.io
NAME                              WEBHOOKS   AGE
validation-tidb-pod-webhook-cfg   1          120m
# helm get hooks tidb-operator 
(empty result)
```
### step 4: Uninstall tidb-operator
```
# helm uninstall tidb-operator
release "tidb-operator" uninstalled
# helm get hooks tidb-operator
Error: release: not found
# kubectl get validatingWebhookConfigurations.admissionregistration.k8s.io
No resources found in default namespace.
```

Reply:
ok